trewrap/src/test/java/org/codice/imaging/nitf/trewrap/SECURA_WrapTest.java

```diff
@@ -0,0 +1,261 @@
+        Tre tre = parseTRE(mTestInvalidCompression, "SECURA");
+        SECURA secura = new SECURA(tre);
+        String error = secura.getValidity().getValidityResultDescription();
+        assertEquals(error, SECURA.INVALID_COMPRESSION);
+    }
+
+
+    public static void checkNitf20SecurityMetadataUnclasAndEmpty(SecurityMetadata securityMetadata) {
+        assertNotNull(securityMetadata);
+        assertEquals(SecurityClassification.UNKNOWN, securityMetadata.getSecurityClassification());
+        assertNull(securityMetadata.getSecurityClassificationSystem());
+        assertEquals("", securityMetadata.getCodewords());
+        assertEquals("", securityMetadata.getControlAndHandling());
+        assertEquals("", securityMetadata.getReleaseInstructions());
+        assertNull(securityMetadata.getDeclassificationType());
+        assertNull(securityMetadata.getDeclassificationDate());
+        assertNull(securityMetadata.getDeclassificationExemption());
+        assertNull(securityMetadata.getDowngrade());
+        assertNull(securityMetadata.getDowngradeDate());
+        assertNull(securityMetadata.getClassificationText());
+        assertNull(securityMetadata.getClassificationAuthorityType());
+        assertEquals("", securityMetadata.getClassificationAuthority());
+        assertNull(securityMetadata.getClassificationReason());
+        assertEquals("", securityMetadata.getSecurityControlNumber());
+    }
+
+    public static void checkNitf21SecurityMetadataUnclasAndEmpty(SecurityMetadata securityMetadata) {
+        Assert.assertEquals(SecurityClassification.UNKNOWN, securityMetadata.getSecurityClassification());
+        assertEquals("", securityMetadata.getSecurityClassificationSystem());
+        assertEquals("", securityMetadata.getCodewords());
+        assertEquals("", securityMetadata.getControlAndHandling());
+        assertEquals("", securityMetadata.getReleaseInstructions());
+        assertEquals("", securityMetadata.getDeclassificationType());
+        assertEquals("", securityMetadata.getDeclassificationDate());
+        assertEquals("", securityMetadata.getDeclassificationExemption());
+        assertEquals("", securityMetadata.getDowngrade());
+        assertEquals("", securityMetadata.getDowngradeDate());
+        assertEquals("", securityMetadata.getClassificationText());
+        assertEquals("", securityMetadata.getClassificationAuthorityType());
+        assertEquals("", securityMetadata.getClassificationAuthority());
+        assertEquals("", securityMetadata.getClassificationReason());
+        assertEquals("", securityMetadata.getSecurityControlNumber());
+    }
+
+    private static byte[] compress(String data) {
+        byte[] compressed = null;
+        try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        GZIPOutputStream gzip = new GZIPOutputStream(bos);) {
+            gzip.write(data.getBytes(StandardCharsets.ISO_8859_1));
```

Comment:
... so when we convert it to a byte array we should be using `StandardCharsets.UTF_8`.

Reply:
changed encoding.

Reply:
Changed to use UTF-8 encoding.